packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+
+// Interfaces
+import { ILivenessModule2 } from "interfaces/safe/ILivenessModule2.sol";
+
+/// @title LivenessModule2
+/// @notice This module allows for challenge-based ownership transfer to a fallback owner
+///         when the Safe becomes unresponsive. The fallback owner can initiate a challenge,
+///         and if the Safe doesn't respond within the challenge period, ownership transfers
+///         to the fallback owner.
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this module using ModuleManager.enableModule()
+///      2. The Safe must then configure the module by calling enableModule() with parameters
+contract LivenessModule2 is ILivenessModule2 {
+    /// @notice Reserved address used as the previous owner to the first owner in a Safe
+    address public constant SENTINEL_OWNER = address(0x1);
+
+    /// @notice Configuration for a Safe's liveness module
+    struct ModuleConfig {
+        uint256 livenessResponsePeriod;
+        address fallbackOwner;
+    }
+
+    /// @notice Mapping from Safe address to its configuration
+    mapping(address => ModuleConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to active challenge start time (0 if no challenge)
+    mapping(address => uint256) public challengeStartTime;
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Enables the module by the multisig to be challenged and sets the liveness_response_period and
+    /// fallback_owner
+    /// @dev MUST set the caller as a safe
+    /// @dev MUST take as parameters liveness_response_period and fallback_owner and store them as related to the safe
+    /// @dev MUST accept an arbitrary number of independent safe contracts to enable the module
+    /// @param _livenessResponsePeriod The period in seconds for a liveness response
+    /// @param _fallbackOwner The address that will become owner if challenge succeeds
+    function enableModule(uint256 _livenessResponsePeriod, address _fallbackOwner) external {
+        if (_livenessResponsePeriod == 0 || _fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidParameters();
+        }
```

Comment:
We also need to require `safe.isModuleEnabled(address(this))`, otherwise you can call this on a module that can't act on the safe.

Which means also that @smartcontracts is right. This needs to be `configure`, because enabling the module means something else.

Reply:
Solved in [37ae803](https://github.com/ethereum-optimism/optimism/pull/17272/commits/37ae803b8080e1b2aa502cd33d3a24d3e0134ed2)